src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -2681,6 +2693,23 @@ master_dist_placement_cache_invalidate(PG_FUNCTION_ARGS)
 
 	CheckCitusVersion(ERROR);
 
+	/*
+	 * Before 7.0-2 this trigger is on pg_dist_shard_placement,
+	 * ignore trigger in this scenario.
+	 */
+	Oid pgDistShardPlacementId = get_relname_relid("pg_dist_shard_placement",
+												   PG_CATALOG_NAMESPACE);
```

Comment:
Two comments that I'm not sure how important they are:
1. Maybe actually just check the version of the citus extension, to make this check a bit less hacky.
2. This is a non cached lookup. I'm not sure how much of a hot-path this is, but maybe this can simply be made cached.

Reply:
1. There's an `InstalledExtensionVersion` function,  but in the end this isn't a very hacky solution since we're casting the tuples to a type that doesn't match `pg_dist_shard_placement`
2. This path isn't hot enough for the lookup to matter